django_activeurl/utils.py

```diff
@@ -52,31 +54,51 @@ def get_cache_key(content, css_class, parent_tag, full_path, menu):
     return cache_key
 
 
-def check_active(url, element, full_path, css_class, menu):
-    '''check "active" url, apply css_class'''
-    # django > 1.5 template boolean\None variables feature
-    if isinstance(menu, bool):
-        if menu:
-            menu = 'yes'
+def yesno_to_bool(value, varname):
+    """Return True/False from "yes"/"no".
+
+    :param value: template keyword argument value
+    :type value: string
+    :param varname: name of the variable, for use on exception raising
+    :type varname: string
+    :raises: :exc:`ImproperlyConfigured`
+
+    Django > 1.5 template boolean/None variables feature.
+    """
+    if isinstance(value, bool):
+        if value:
+            value = 'yes'
         else:
-            menu = 'no'
-    elif menu is None:
-        menu = 'no'
-    # check menu configuration, set boolean value
-    if menu.lower() in ('yes', 'true'):
-        menu = True
-    elif menu.lower() in ('no', 'false'):
-        menu = False
+            value = 'no'
+    elif value is None:
+        value = 'no'
+
+    # check value configuration, set boolean value
+    if value.lower() in ('yes', 'true'):
+        value = True
+    elif value.lower() in ('no', 'false'):
+        value = False
     else:
-        raise ImproperlyConfigured('''
-            malformed menu value
-        ''')
+        raise ImproperlyConfigured(
+            'activeurl: malformed param value for %s' % varname
+        )
+    return value
+
+
+def check_active(url, element, full_path, css_class, menu, ignore_params):
+    '''check "active" url, apply css_class'''
+    menu = yesno_to_bool(menu, 'menu')
+    ignore_params = yesno_to_bool(ignore_params, 'ignore_params')
+
     # check missing href parameter
     if not url.attrib.get('href', None) is None:
         # get href attribute
         href = url.attrib['href'].strip()
         # cut off hashtag (anchor)
         href = re.sub(r'\#.+', '', href)
+        if ignore_params:
```

Comment:
This is actually less performant IMO, but I added it in 21ce8a3.  While it feels good to work with the standard library.

1. it turns out urlparse objects aren't mutable. The reason why is the object subclasses ``tuple``.
2. ``_replace`` is available, but it does a copy every time

So don't you think ``re.sub`` would be more performant?